code/game/objects/structures/flora.dm

```diff
@@ -142,10 +142,116 @@
 	..()
 	var/number = rand(1,36)
 	if (number == 36)
-		if (prob(90))//Make the wierd one rarer
+		if (prob(90))//Make the weird one rarer
 			number = rand(1,35)
 		else
-			desc = "It stares into your soul."
+			desc = "A half-sentient plant borne from a mishap in a Zeng-Hu genetics lab."
+
+	if (number == 1)

```

Comment:
this chain of if statements is really ugly. make it into switch statements

```
switch(number)
    if(1)
          desc = "Just your common, everyday houseplant." // cool flavortext for plants
    if(2)
          desc = "Just your common, everyday houseplant."
```

And just follow that pattern.

Reply:
Also I noticed that most of them have literally same description `desc = "Just your common, everyday houseplant."`. Please make special cases as `if` and anything else can go into `else`

Reply:
So remove 1, 2, 6, 8, 10, 11, 12, 16, 17, 21, 23, 24, 25, 26, 28, 29, 30, 31, 35. Leave the rest into `switch(number)` statement as ifs and like things above that I mention will go into 
```
else
    desc = "Just your common, everyday houseplant."
```

Reply:
Ah, thanks Sindorman. I was trying to figure that out for the longest time until I gave up and brute-forced it.

Reply:
Fixed.